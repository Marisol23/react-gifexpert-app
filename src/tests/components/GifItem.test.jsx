import { render, screen } from '@testing-library/react'
import { GifItem } from "../../components/GifItem"

describe('Test in <GifItem/>', () => {

    const title = 'One Punch'
    const url = 'https://one-punch.com/one-puch.jpg'

    test('debe hacer match con el snapshot ', () => {
        
        const {container} = render(<GifItem title={title} url={url}/>)
        expect(container).toMatchSnapshot();
      
    }),

    test('debe mostrar la imagen con el URL y el ALT iNDICADO', () => {
        render(<GifItem title={title} url={url}/>);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(src);
        expect(alt).toBe(alt);
    }),
    test('should ', () => {
        render(<GifItem title={title} url={url}/>)
        expect(screen.getByText(title)).toBeTruthy();
    }) 
})
