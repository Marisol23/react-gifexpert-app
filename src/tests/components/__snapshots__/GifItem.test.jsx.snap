// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifItem/> debe hacer match con el snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch"
      src="https://one-punch.com/one-puch.jpg"
    />
    <p>
      One Punch
    </p>
  </div>
</div>
`;
